pkgs/games/steam/fhsenv.nix

```
@@ -279,7 +293,8 @@ in buildFHSEnv rec {
     exec steam ${extraArgs} "$@"
   '';
 
-  inherit extraBwrapArgs;
+  extraBwrapArgs = extraBwrapArgs-modified;
+  extraPreBwrapCmds = extraPreBwrapCmds-modified;
```

Could probably just inline these, couldn't we?

---

Nope. The attrset is a `rec`, so that would create infinite recursion because the variable names were shadowed. Like I said, it's a lexical scoping minefield trying to do this reasonably. I don't like this, either, I just don't know how to do it better... :frowning:

---

Ah, I see the problem. The usual solution to this is the `args@` pattern or `finalAttrs` pattern. Not sure the latter would work here without unreasonable effort.

```suggestion
  extraBwrapArgs = "..." + args.extraBwrapArgs;
  extraPreBwrapCmds = "..." + args.extraPreBwrapCmds;
```

---

Ah, good idea. Don't know why I didn't think of that.

---

Turns out that's annoying in a new way... the defaults set at the binding site don't affect `args`, so you have to provide the default again at the use site...